File: src/diff_sum_gen.rs
use crate::{helper, llm};

const DEFAULT_PROMPT: &str = "summarize the git diff changes.
List the key modifications, what was added, removed, or modified, and briefly explain their purpose or impact if possible.
about only changes. must not write about project. you don't readme writer, you summarize diff changes.";

pub async fn sum_diff<T: AsRef<str>>(
    diff: T,
    status: T,
    model: llm::LlmReqInfo,
    lang: Option<T>,
    extra: Option<T>,
) -> Result<String, llm::Error> {
    let lang = helper::init_lang(lang);
    let st = status.as_ref();
    let extra = helper::init_extra(extra);
    let diff = diff.as_ref();
    let prompt =
        format!("Please in {lang}.\n{DEFAULT_PROMPT}\nstatus: {st}\ndiff: {diff}.\n{extra}");

    llm::call_llm(model, prompt).await
}

File: ggw/diff_sum_gen.py
"""Diff summarization."""
from __future__ import annotations

from typing import Optional

from ggw.llm import LlmReqInfo, call_llm

_DEFAULT_PROMPT = (
    "summarize the git diff changes.\n"
    "List the key modifications, what was added, removed, or modified,"
    " and briefly explain their purpose or impact if possible.\n"
    "about only changes. must not write about project."
    " you don't readme writer, you summarize diff changes."
)


def sum_diff(
    diff: str,
    status: str,
    model: LlmReqInfo,
    lang: Optional[str] = None,
    extra: Optional[str] = None,
) -> str:
    lang = lang or "english"
    extra_str = f" # Additional Instructions: {extra}" if extra else ""
    prompt = (
        f"Please in {lang}.\n{_DEFAULT_PROMPT}\n"
        f"status: {status}\ndiff: {diff}.\n{extra_str}"
    )
    return call_llm(model, prompt)
